Fix documentation for additionalRuntimeClasspath for <=1.21.9 (#357)
neoforged/Documentation: toolchain/docs/dependencies/nonmclibs.md

    additionalRuntimeClasspath 'com.example:example:1.0'
}
```

> :::warning

Reviewer: There should be a newline before this.

> However, using a library that has transitive dependencies will require you to manually exclude and `additionalRuntimeClasspath` / `implementation` the dependency yourself. This is only needed if that specific module is required at runtime; compilation will proceed fine.

Reviewer: ```suggestion
If the library contains any transitive dependencies required at runtime, the transitive dependencies have to be manually excluded and readded to the `additionalRuntimeClasspath` configuration.
```

Reworded it. I took out the `implementation` since compile time should handle that without issue.

```gradle
dependencies {
    // This is required to add the library at compile time
    implementation 'com.example:example:1.0'
    // This adds the library to all the runs
    additionalRuntimeClasspath 'com.example:example:1.0' {

>         // exclude the transitive dependency you want to add to the classpath

Reviewer: ```suggestion
        // Exclude the transitive dependency that is required at runtime
```

Just being more explicit.

        exclude 'com.example:dependency'
    }
    // add the module back to the classpath

>     additionalRuntimeClasspath 'com.example:my_module:1.0'

Reviewer: ```suggestion
    // Add the module back to the runtime classpath
    additionalRuntimeClasspath 'com.example:dependency:1.0'
```

Missed `my_module` change.
